zkevm-circuits/src/rlp_decoder_tables.rs

```diff
@@ -0,0 +1,764 @@
+            },
+        };
+        (rlp_type, decodable)
+    }
+}
+
+/// rules of tx members
+pub const RLP_TX_FIELD_DECODE_RULES: [(RlpTxTypeTag, RlpTxFieldTag, RlpDecodeRule); 14] = [
+    (
+        RlpTxTypeTag::TxLegacyType,
+        RlpTxFieldTag::TxListRlpHeader,
+        RlpDecodeRule::LongList,
+    ),
+    (
+        RlpTxTypeTag::TxLegacyType,
+        RlpTxFieldTag::TxRlpHeader,
+        RlpDecodeRule::LongList,
+    ),
+    (
+        RlpTxTypeTag::TxLegacyType,
+        RlpTxFieldTag::Nonce,
+        RlpDecodeRule::Uint96,
+    ),
+    (
+        RlpTxTypeTag::TxLegacyType,
+        RlpTxFieldTag::GasPrice,
+        RlpDecodeRule::Uint96,
+    ),
+    (
+        RlpTxTypeTag::TxLegacyType,
+        RlpTxFieldTag::Gas,
+        RlpDecodeRule::Uint96,
+    ),
+    (
+        RlpTxTypeTag::TxLegacyType,
+        RlpTxFieldTag::To,
+        RlpDecodeRule::Address,
+    ),
+    (
+        RlpTxTypeTag::TxLegacyType,
+        RlpTxFieldTag::To,
+        RlpDecodeRule::Empty,
+    ),
+    (
+        RlpTxTypeTag::TxLegacyType,
+        RlpTxFieldTag::Value,
+        RlpDecodeRule::Uint96,
+    ),
+    (
+        RlpTxTypeTag::TxLegacyType,
+        RlpTxFieldTag::Data,
+        RlpDecodeRule::Bytes48K,
+    ),
+    (
+        RlpTxTypeTag::TxLegacyType,
+        RlpTxFieldTag::SignV,
+        RlpDecodeRule::Uint96,
+    ),
+    (
+        RlpTxTypeTag::TxLegacyType,
+        RlpTxFieldTag::SignR,
+        RlpDecodeRule::Uint256,
+    ),
+    (
+        RlpTxTypeTag::TxLegacyType,
+        RlpTxFieldTag::SignS,
+        RlpDecodeRule::Uint256,
+    ),
+    (
+        RlpTxTypeTag::TxLegacyType,
+        RlpTxFieldTag::Padding,
+        RlpDecodeRule::Padding,
+    ),
+    (
+        RlpTxTypeTag::TxLegacyType,
+        RlpTxFieldTag::DecodeError,
+        RlpDecodeRule::Padding,
+    ),
+];
+
+/// Table that contains the fields of all possible RLP decodable fields
+#[derive(Clone, Debug)]
+pub struct RlpDecoderTable {
+    /// The length of the field
+    pub table_tag: Column<Fixed>,
+    /// The tx type tag
+    pub tx_type: Column<Fixed>,
+    /// The tx field tag
+    pub tx_field_tag: Column<Fixed>,
+    /// The RLP type
+    pub rlp_type: Column<Fixed>,
+    /// The first byte of the RLP encoded field
+    pub byte_0: Column<Fixed>,
+    /// Whether the field is decodable
+    pub decodable: Column<Fixed>,
+}
+
+impl RlpDecoderTable {
+    /// Construct a new RlpDecoderTable
+    pub fn construct<F: Field>(meta: &mut ConstraintSystem<F>) -> Self {
+        Self {
+            table_tag: meta.fixed_column(),
+            tx_type: meta.fixed_column(),
+            tx_field_tag: meta.fixed_column(),
+            rlp_type: meta.fixed_column(),
+            byte_0: meta.fixed_column(),
+            decodable: meta.fixed_column(),
+        }
+    }
+
+    /// build from existed columns
+    pub fn build_from_columns(columns: &[Column<Fixed>]) -> Self {
+        assert!(columns.len() > 5);
```

Comment:
This one is to make sure developer (me particularly) choose a big enough N in `impl<const N: usize> RlpDecoderFixedTable<N>`, if N is smaller than required, it panics immediately, then the developer could choose a bigger one to make code run.
Normally malicious entity does not make panic this way (but surely lots of other ways, what a mess :( ), because it impacts itself only, others still could give correct proof.